docs/proposals/20241127-namespace-separation.md

```diff
@@ -0,0 +1,390 @@
+
+### Goals
+There are some restrictions while using multiple providers, see: https://cluster-api.sigs.k8s.io/developer/core/support-multiple-instances
+But we need to:
+
+1. extend the existing "cache configuration" feature `--namespace=<ns>` (limit watching a single namespace) to limit watching on multiple namespaces.
+2. add new feature to watch on all namespaces except selected ones.
+3. then we run multiple CAPI controller instances:
+   - each watching only specified namespaces: `capi1-system`, …, `capi$(N-1)-system`
+   - and the last resort instance to watch the rest of namespaces excluding the namespaces already watched by previously mentioned instances
+
+### Non-Goals/Future Work
+Non-goals:
+
+* It's not necessary to work with all versions of CAPI/CRDs, we consider:
+  * All instances share the same CRDs.
+  * All instances must be compatible with the deployed CRDs.
+* The contract and RBAC need to be solved on specific provider (AWS, AZURE, ...)
+* It is not supported to configure two instances to watch the same namespace:
+  * the `--namespace=nsX` option can only be used by a single instance.
+* if the same namespace is both excluded and included by the same instance, it will be excluded from being watched by that instance.
+
+
+## Proposal
+We are proposing to:
+
+* enable to select multiple namespaces: add `--namespace=<ns1, …>` to extend `--namespace=<ns>` to watch on selected namespaces
+  * the code change involves extending an existing hash to accommodate multiple items.
+  * This change is only a small and straightforward update of the existing feature to limit watching on specified namespace. The maintenance complexity shouldn't be extended here
+* add the new commandline option `--excluded-namespace=<ens1, …>` to define list of excluded namespaces
+  * the code [change](https://github.com/kubernetes-sigs/cluster-api/pull/11370/files#diff-c4604297ff388834dc8c6de30c847f50548cd6dd4b2f546c433b234a27ad4631R263) is only setting an option `Cache.Options.DefaultFieldSelector` to disable matching with any of specified namespace's names
+  * the maintenance complexity shouldn't be extended a lot here
+
+Note:
+
+* There is also the existing `--watch-filter=<...>` option, which is used for event filtering, whereas `--namespace=<...>` and `--excluded-namespace=<...>` configure the cache.
+
+Our objectives include:
+
+- Each CAPI instance runs in a separate namespace and uses its own service account.
+  - Namespaces can be specified through command-line arguments:
+    - To watch: e.g., `--namespace <ns1> --namespace <ns2>`
+    - To exclude from watching: e.g., `--excluded-namespace <ns1> --excluded-namespace <ns2>`
+- We don't need to support for multiple CAPI versions, but:
+  - All instances must be compatible with the deployed CRDs.
+  - CRDs are deployed only for the newest CAPI instance (selecting one instance with the newest version).
+  - All conversion webhooks in CRDs point to the newest CAPI instance.
```

Comment:
It would probably be safer to assert that CRDs and their conversion are handled separately (as a separate binary) rather than running them alongside controllers and potentially "moving around" if one becomes newer